inline.go

```diff
@@ -212,6 +212,11 @@ func link(p *parser, out *bytes.Buffer, data []byte, offset int) int {
 	var t linkType
 	if offset > 0 && data[offset-1] == '!' {
 		t = linkImg
+		// if footnotes extension is ON and we've seen "![^", then it's not an
+		// image, it's a deferred footnote:
+		if p.flags&EXTENSION_FOOTNOTES != 0 && len(data)-1 > offset && data[offset+1] == '^' {
+			t = linkDeferredFootnote
+		}
```

Comment:
There is a comment right above:

```Go
// [text] == regular link
// ![alt] == image
// ^[text] == inline footnote
// [^refId] == deferred footnote
```

I wonder if it's worth adding this special case to that list, i.e.:

```Go
// ![^refId] == deferred footnote
```